### PR TITLE
Dont call as_json on Delayed::Backend::Sequel::Job

### PR DESCRIPTION
**Pull request by davidbegin**
instead call #to_hash, and turn all the keys to strings, to
  keep the rest of the implementation untouched

### CODE REVIEW

**Comment by davidbegin** [comment]
We are seeing an error in Rollbar, when using Sequel and Delayed Job.
'undefined method `as_json'' for #<Delayed::Backend::Sequel::Job:0x007fe7f57d31f8>

In this file: https://github.com/rollbar/rollbar-gem/blob/master/lib/rollbar/plugins/delayed_job/job_data.rb#L9

we are calling on @as_json on this class in Delayed Job Sequel
https://github.com/TalentBox/delayed_job_sequel/blob/56e7d69e1710b75ae7044a884961e1ebfd6a4661/lib/delayed/backend/sequel.rb

I updated Rollbar to just use #to_hash, if #as_json is not defined.

**Comment by jondeandres** [comment]
Hey @davidbegin, thanks for this. Let me test it and we'll merge it.

File: lib/rollbar/plugins/delayed_job/job_data.rb [hunk 1]
```diff
@@ -6,7 +6,12 @@ def initialize(job)
   end
 
   def to_hash
-    job_data = job.as_json
+    job_data = if job.respond_to?(:as_json)
```
Comment (jondeandres) [inline lib/rollbar/plugins/delayed_job/job_data.rb]: I think we should extract this code into its own method. And also, we should ensure that `job` responds to `#to_hash`.

Comment (davidbegin) [inline lib/rollbar/plugins/delayed_job/job_data.rb]: Thanks for reviewing this, I will work on those changes right now.

**Comment by jondeandres** [comment]
Hey @davidbegin,

thank you very much for the notes. You've found a missing piece we forgot to move when we added the plugins architecture. So, `JobData` should be in the correct namespace and the `require` in the job_data spec should be fixed.

We'll do this change ourself and check that the specs are passing in master. After that, I think it's a good starting for your PR.

**Comment by michaelirey** [comment]
@jondeandres Has there been any movement on this PR? Is there anything I could do to help?

Thanks!

**Pull request closed without merging**